### PR TITLE
feat(condo): INFRA-379 add cache for access of the resident and service consumer

### PR DESCRIPTION
**Pull request by sitozzz**

### CODE REVIEW

File: apps/condo/domains/resident/utils/accessSchema.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+const conf = require('@open-condo/config')
+const { getRedisClient } = require('@open-condo/keystone/redis')
+const { find } = require('@open-condo/keystone/schema')
+
+const _redisClient = getRedisClient('default', 'cache')
+const DEFAULT_CACHE_TTL_IN_MS = 60 * 60 * 1000 // 1 hour
+const CACHE_TTL_FROM_ENV = parseInt(conf['USER_RESIDENT_CACHING_TTL_IN_MS'])
+const CACHE_TTL_IN_MS = isNaN(CACHE_TTL_FROM_ENV) ? DEFAULT_CACHE_TTL_IN_MS : CACHE_TTL_FROM_ENV
+const DISABLE_USER_RESIDENT_CACHING = get(conf, 'DISABLE_USER_RESIDENT_CACHING', 'false').toLowerCase() === 'true'
```
Comment (toplenboren) [inline apps/condo/domains/resident/utils/accessSchema.js:11]: I think our common pattern of configuring stuff was JSON

I expected smth like: RESIDENT_CACHE_CONFIG=<JSON>

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/resident/utils/accessSchema.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+const { get, set } = require('lodash')
+
+const conf = require('@open-condo/config')
+const { getRedisClient } = require('@open-condo/keystone/redis')
+const { find } = require('@open-condo/keystone/schema')
+
+const _redisClient = getRedisClient('default', 'cache')
+const DEFAULT_CACHE_TTL_IN_MS = 60 * 60 * 1000 // 1 hour
```
Comment (SavelevMatthew) [inline apps/condo/domains/resident/utils/accessSchema.js:8]: We don't migrate resident, so it has no problem with increasing TTL, like in organisations. But I guess average mobile session is under 1 hour, so it's oke

**Review by SavelevMatthew — COMMENTED** [review]
Have no time to review it fully, but have 2 core comments:
1. You should limit, which fields are going to cache
2. You should add JSDocs, so everyone using your utils can understand, what's inside cache

**Comment by sitozzz** [comment]
> Have no time to review it fully, but have 2 core comments:
> 
> 1. You should limit, which fields are going to cache
> 2. You should add JSDocs, so everyone using your utils can understand, what's inside cache

So I just done that. Please check it out

File: apps/condo/domains/resident/utils/accessSchema.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+/**
+ * Obtains user resident info via caching or sub-querying
+ * @param {{ req: import('express').Request }} ctx - keystone context
+ * @param {{ id: string }} user - user object
+ * @returns {Promise<Array<ResidentCache>>}
+ */
+async function getUserResidents (ctx, user) {
+    return (await _getUserResidents(ctx, user)).residents || []
```
Comment (SavelevMatthew) [inline apps/condo/domains/resident/utils/accessSchema.js:90]: Why do you use `|| []` here? Seems like its always contains arrays, isn't it?

File: apps/condo/domains/resident/utils/accessSchema.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+const { getRedisClient } = require('@open-condo/keystone/redis')
+const { find } = require('@open-condo/keystone/schema')
+
+const _redisClient = getRedisClient('default', 'cache')
+const DEFAULT_CACHE_TTL_IN_MS = 60 * 60 * 1000 // 1 hour
+const CACHE_TTL_FROM_ENV = parseInt(conf['USER_RESIDENT_CACHING_TTL_IN_MS'])
+const CACHE_TTL_IN_MS = isNaN(CACHE_TTL_FROM_ENV) ? DEFAULT_CACHE_TTL_IN_MS : CACHE_TTL_FROM_ENV
+const DISABLE_USER_RESIDENT_CACHING = get(conf, 'DISABLE_USER_RESIDENT_CACHING', 'false').toLowerCase() === 'true'
+const RESIDENT_CACHE_FIELDS = ['id', 'unitName', 'unitType', 'addressKey', 'organization', 'property']
+const SERVICE_CONSUMER_CACHE_FIELDS = ['id', 'organization', 'accountNumber']
```
[the remark this answers is not in the document]
Comment (sitozzz) [reply apps/condo/domains/resident/utils/accessSchema.js:13]: For now that isn't necessary

File: apps/condo/domains/resident/utils/accessSchema.js (hunk 3 above)
Comment (sitozzz) [reply apps/condo/domains/resident/utils/accessSchema.js:90]: Old one. Fixed that

**Review by SavelevMatthew — APPROVED** [review]

**Pull request merged**